tests/test_norm_ops.py

```diff
@@ -136,6 +136,39 @@ def test_accuracy_layernorm(shape, dtype):
     gems_assert_close(res_bias_grad, ref_bias_grad, dtype, reduce_dim=M)
 
 
+@pytest.mark.weight_norm_interface
+@pytest.mark.parametrize("shape", REDUCTION_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("dim", [0, -1])
+def test_accuracy_weightnorm(shape, dtype, dim):
+    dim = (dim - len(shape)) % len(shape)
+    v = torch.randn(shape, dtype=dtype, device="cuda", requires_grad=True)
+    g = torch.randn(shape[dim], dtype=dtype, device="cuda", requires_grad=True)
```

Comment:
If I change this line to `g = torch.randn((1,), dtype=dtype, device="cuda", requires_grad=True)`, I will receive a grad assert error. Is this in line with expectations?

Reply:
The size of g must be greater than or equal to shape[dim]; other cases are unpredictable. Refer to [this line](https://github.com/pytorch/pytorch/blob/9529d018e937b8b2a53cccc72f38be283828715a/aten/src/ATen/native/cuda/WeightNorm.cu#L140), where you can see that g does not have out-of-bounds handling.